typescript/ecs/capacity-providers/index.ts

```diff
@@ -0,0 +1,166 @@
+import ecs = require('@aws-cdk/aws-ecs');
+import ec2 = require('@aws-cdk/aws-ec2');
+import cdk = require('@aws-cdk/core');
+import elb = require('@aws-cdk/aws-elasticloadbalancingv2');
+
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'aws-ecs-demo-capacity-providers');
+
+// Create cluster with FG capacityProviders enabled
+const vpc = new ec2.Vpc(stack, 'Vpc', { maxAzs: 2 });
+const cluster = new ecs.Cluster(stack, 'FargateCPCluster', {
+  vpc,
+  capacityProviders: ['FARGATE', 'FARGATE_SPOT'],
+});
+
+// Create task def with sample image
+const taskDefinition = new ecs.FargateTaskDefinition(stack, 'TaskDef');
+const container = taskDefinition.addContainer('web', {
+  image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample"),
+  memoryLimitMiB: 256,
+});
+
+container.addPortMappings({
+  containerPort: 80,
+  protocol: ecs.Protocol.TCP
+});
+
+/*
+ * The following service examples all have `desiredCount` set to be an integer. We examine how to satisfy different user
+ * stories that leverage capacity providers with a fixed `desiredCount` using the existing CFN API:
+ *
+ * NOTE: The current behavior of the service if it cannot find spot capacity to place tasks on is to simply fail. For
+ * that reason, any specification for placement on spot should not be treated as a guarantee.
+ *
+ * Case 1: m services split across both spot and on-demand
+ * Case 2: m services placed on spot only (100% spot)
+ * Case 3: n services first placed on spot, remaining m-n placed on on-demand
+ * Case 4: n services first placed on on-demand, remaining m-n placed on spot
+ *
+ * The case of all m services being placed on on-demand capacity is equivalent to using the fargate launchType without
+ * capacity providers at all, so will not be considered in this example.
+ */
+
+ // Case 1: m services split across both spot and on-demand
+ 
+const service1 = new ecs.FargateService(stack, 'FargateService-NoBase-EqualWeight', {
+  cluster,
+  taskDefinition,
+  desiredCount: 10,
+  capacityProviderStrategies: [
+    {
+      capacityProvider: 'FARGATE_SPOT',
+      weight: 1,
+    },
+    {
+      capacityProvider: 'FARGATE',
+      weight: 1,
+    },
```

Comment:
Will it prefer spot by default with the same weight on these? I'd imagine splitting across spot with a desired count what users want is use spot if available, otherwise fargate